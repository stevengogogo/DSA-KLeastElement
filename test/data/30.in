10 50000
8996 7644 7862 -8635 -5746 -3842 7503 -3388 7622 -663
Reverse 5 9
Delete 5
Reverse 8 8
Insert 4 -9558
Query 2 10 2
Reverse 9 10
Query 9 9 1
Insert 3 -6618
Delete 2
Insert 6 -6136
Delete 9
Reverse 8 9
Delete 7
Reverse 1 6
Delete 3
Delete 5
Query 2 5 1
Insert 4 1795
Query 4 5 2
Query 7 7 1
Query 7 7 1
Delete 8
Insert 5 -7322
Reverse 7 7
Query 4 5 2
Query 1 2 2
Insert 2 1373
Reverse 8 8
Insert 8 -6165
Insert 9 5496
Reverse 2 5
Delete 6
Query 8 8 1
Delete 7
Query 1 7 7
Reverse 4 8
Insert 1 -8505
Delete 2
Insert 7 -2832
Reverse 4 6
Reverse 7 8
Reverse 6 6
Reverse 4 9
Delete 3
Insert 8 209
Reverse 7 10
Delete 7
Reverse 5 6
Query 1 9 1
Query 5 6 1
Delete 5
Delete 3
Query 4 6 3
Reverse 4 6
Insert 7 -8572
Delete 5
Query 6 6 1
Query 3 6 4
Reverse 1 3
Reverse 1 6
Reverse 5 6
Insert 7 -2660
Query 8 8 1
Reverse 7 7
Insert 1 9975
Insert 2 -9408
Delete 2
Insert 3 907
Insert 7 2955
Query 10 10 1
Reverse 1 8
Reverse 11 11
Query 5 8 1
Insert 6 -1013
Query 9 10 1
Delete 5
Delete 8
Reverse 5 6
Delete 6
Reverse 5 8
Reverse 5 8
Delete 6
Query 8 8 1
Insert 1 -8424
Query 5 7 2
Delete 6
Reverse 2 3
Reverse 4 4
Insert 7 7392
Query 4 7 1
Delete 9
Insert 3 2498
Reverse 1 3
Insert 4 -3949
Delete 5
Delete 8
Insert 1 5573
Insert 8 -4890
Delete 10
Delete 4
Query 2 4 3
Reverse 8 8
Reverse 3 8
Reverse 5 6
Insert 5 -9307
Delete 8
Reverse 2 3
Insert 5 -8898
Reverse 5 9
Query 6 8 2
Delete 8
Query 6 7 2
Insert 4 -808
Insert 2 -4713
Reverse 1 10
Query 7 10 1
Query 5 10 5
Insert 11 -6200
Query 6 10 5
Delete 6
Delete 10
Query 4 6 1
Query 6 6 1
Query 2 7 4
Reverse 2 7
Delete 7
Delete 7
Reverse 2 5
Insert 3 2435
Query 8 8 1
Delete 8
Delete 5
Reverse 1 1
Query 5 5 1
Reverse 3 6
Insert 6 -3896
Insert 4 -1325
Reverse 8 8
Query 5 8 2
Delete 6
Insert 4 728
Insert 2 -4524
Reverse 1 2
Delete 6
Query 6 6 1
Delete 7
Query 4 7 3
Reverse 3 7
Reverse 3 6
Reverse 2 5
Query 7 7 1
Insert 7 2147
Reverse 4 6
Delete 3
Insert 6 -1257
Query 5 7 3
Reverse 6 6
Query 8 8 1
Query 1 1 1
Query 5 8 4
Delete 4
Query 4 4 1
Reverse 1 4
Delete 4
Insert 5 -9491
Delete 3
Reverse 1 3
Delete 2
Reverse 1 5
Insert 6 -7344
Reverse 6 6
Reverse 3 3
Delete 1
Reverse 5 5
Insert 6 -893
Delete 1
Query 1 3 2
Query 2 2 1
Query 4 5 1
Insert 1 8233
Reverse 2 2
Query 5 5 1
Reverse 4 6
Query 2 3 1
Query 4 5 1
Query 1 3 2
Delete 6
Delete 4
Delete 4
Query 2 2 1
Insert 2 3750
Delete 2
Reverse 3 3
Insert 1 -387
Delete 3
Delete 3
Delete 1
Reverse 1 1
Query 1 1 1
Delete 1
Insert 1 -5424
Reverse 1 1
Delete 1
