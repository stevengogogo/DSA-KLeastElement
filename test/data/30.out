-8635
-663
-8635
1795
7503
7503
1795
-6136
5496
7862
-8635
-663
1373
-8572
1373
5496
-2660
-8572
1795
5496
907
-8505
2955
-8505
209
-4713
2498
5573
-4890
2498
1795
5573
1795
-3896
1795
1795
2955
2147
2955
-4524
2955
209
-1257
209
-7344
-7344
-9491
-7344
209
-9491
8233
